_ml-commons-plugin/algorithms.md

```diff
@@ -0,0 +1,315 @@
+
+**Request**
+
+```json
+POST _plugins/_ml/_predict/LINEAR_REGRESSION/ROZs-38Br5eVE0lTsoD9
+{
+  "parameters": {
+    "target": "price"
+  },
+  "input_data": {
+    "column_metas": [
+      {
+        "name": "A",
+        "column_type": "DOUBLE"
+      },
+      {
+        "name": "B",
+        "column_type": "DOUBLE"
+      }
+    ],
+    "rows": [
+      {
+        "values": [
+          {
+            "column_type": "DOUBLE",
+            "value": 3
+          },
+          {
+            "column_type": "DOUBLE",
+            "value": 5
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+**Response**
+
+```json
+{
+  "status": "COMPLETED",
+  "prediction_result": {
+    "column_metas": [
+      {
+        "name": "price",
+        "column_type": "DOUBLE"
+      }
+    ],
+    "rows": [
+      {
+        "values": [
+          {
+            "column_type": "DOUBLE",
+            "value": 17.25701855310131
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+### Limitations
+
+ML Commons only supports the linear Stochastic gradient trainer or optimizer, which cannot effectively map the non-linear relationships in trained data. When used with complicated datasets, the linear Stochastic trainer might cause some convergence problems and inaccurate results. 
+
+## RCF
+
+[Random Cut Forest](https://github.com/aws/random-cut-forest-by-aws) (RCF) is a probabilistic data structure used primarily for unsupervised anomaly detection. Its use also extends to density estimation and forecasting. OpenSearch leverages RCF for anomaly detection. ML Commons supports two new variants of RCF for different use cases:
+
+* Batch RCF: Detects anomalies in non-time series data. 
+* Fixed in time (FIT) RCF: Detects anomalies in time series data.
+
+### Parameters
+
+#### Batch RCF
+
+Parameter | Type   | Description | Default Value
+:--- |:--- | :--- | :---
+number_of_trees | integer | The number of trees in the forest | 30
+sample_size | integer | The same size used by the stream samplers in the forest | 256
+output_after | integer | The number of points required by stream samplers before results return | 32
+training_data_size | integer | The size of your training data | Dataset size
+anomaly_score_threshold | double | The threshold of the anomaly score | 1.0 
+
+#### Fit RCF
+
+Parameter | Type   | Description | Default Value
+:--- |:--- | :--- | :---
+number_of_trees | integer | The number of trees in the forest | 30
+shingle_size | integer | A shingle, or a consecutive sequence of the most recent records | 8
+sample_size | integer | The sample size used by stream samplers in the forest | 256
+output_after | integer | The number of points required by stream samplers before results return | 32
+time_decay | double | The decay factor used by stream samplers in the forest | 0.0001 
+anomaly_rate | double | The anomaly rate | 0.005
+time_field | string | (**Required**) The time filed for RCF to use as time series data | N/A
```

Comment:
Since this one is required, I'm guessing all the others are optional?

Reply:
Correct